### ISSUE TITLE
filter-deps seems to be causing trouble for components in X11

### ISSUE THREAD

**Issue by akesandgren** [hpcugent/easybuild-easyconfigs#3803]
In X11-20160819-intel-2016b.eb there is a dependency on ('freetype', '2.6.5'), which in turn depends on zlib. However, with --filter-deps=zlib building of the libXft component fails with:
==========
checking for FREETYPE... configure: error: Package requirements (freetype2 >= 2.1.6) were not met:

Package 'zlib', required by 'freetype2', not found
==========
during configure of libXft. 

At least I assume it's due to filter-deps...

**Comment by boegel** [comment]
@akesandgren well, if you use `--filter-deps=zlib`, you need to make sure `zlib(-devel)` is provided by the OS...

**Comment by akesandgren** [comment]
It is...

**Comment by boegel** [comment]
@akesandgren well, ok, then you'll need to dive in deeper and see why `freetype` of `libXft` thinks that it's not

**Comment by akesandgren** [comment]
Yeah, figured as much, just wanted to have somewhere to discuss stuff :-)

**Comment by akesandgren** [comment]
Ahhh, the problem is probably the EB built pkg-config that is used in the X11 chain.
It's no aware of the debian/ubuntu dir for pkg-config files.

Oh well... guess i have to redo stuff without filter-deps then.

**Comment by akesandgren** [comment]
This is where a site-override file would come in handy...

**Comment by boegel** [comment]
@akesandgren you could add `pkg-config` to `--filter-deps` too, to make EasyBuild pick up the one provides from the OS?
